narrowExceptionVariableInCatchClause.ts(11,17): error TS2551: Property 'doPanic' does not exist on type '{ type: "foo"; dontPanic(): any; }'. Did you mean 'dontPanic'?
narrowExceptionVariableInCatchClause.ts(16,17): error TS2551: Property 'massage' does not exist on type 'Error'. Did you mean 'message'?


==== narrowExceptionVariableInCatchClause.ts (2 errors) ====
    declare function isFooError(x: any): x is { type: 'foo'; dontPanic(); };
    
    function tryCatch() {
        try {
            // do stuff...
        }
        catch (err) { // err is implicitly 'any' and cannot be annotated
    
            if (isFooError(err)) {
                err.dontPanic(); // OK
                err.doPanic(); // ERROR: Property 'doPanic' does not exist on type '{...}'
                    ~~~~~~~
!!! error TS2551: Property 'doPanic' does not exist on type '{ type: "foo"; dontPanic(): any; }'. Did you mean 'dontPanic'?
!!! related TS2728 narrowExceptionVariableInCatchClause.ts:1:58: 'dontPanic' is declared here.
            }
    
            else if (err instanceof Error) {
                err.message;
                err.massage; // ERROR: Property 'massage' does not exist on type 'Error'
                    ~~~~~~~
!!! error TS2551: Property 'massage' does not exist on type 'Error'. Did you mean 'message'?
!!! related TS2728 lib.es5.d.ts:--:--: 'message' is declared here.
            }
    
            else {
                throw err;
            }
        }
    }
    